### PR TITLE
Always push OCI images on azure docker registry

### PR DESCRIPTION
**Pull request by jbern0rd**

### CODE REVIEW

**Review by mcornaton — APPROVED** [review]
Looks great, thanks!

File: build.gradle [hunk 1]
```diff
@@ -33,33 +33,40 @@ repositories {
     maven { url 'https://jitpack.io' }
 }
 
+sourceSets {
+    itest {
+        compileClasspath += sourceSets.main.output
+        runtimeClasspath += sourceSets.main.output
```
Comment (zguesmi) [inline build.gradle:39]:
```
            srcDir 'src/itest/java'
```
?

File: build.gradle [hunk 2]
```diff
@@ -140,22 +147,10 @@ task buildImage {
...
-    }
-}
-
 task itest(type: Test) {
     group 'Verification'
     description 'Runs the integration tests.'
-    dependsOn   buildImage
-    setTestClassesDirs(sourceSets.integrationTest.output)
-    classpath = sourceSets.integrationTest.runtimeClasspath
-    useJUnitPlatform()
```
Comment (zguesmi) [inline build.gradle:160]: Does `itest` still use JUnit 5 without this line?

File: build.gradle (hunk 1 above)
Comment (zguesmi) [reply build.gradle:39]: Nice!

File: build.gradle (hunk 2 above)
Comment (jbern0rd) [reply build.gradle:160]: The shared configuration is:
* we use JUnit5
* a coverage report will be generated after

Comment (zguesmi) [reply build.gradle:160]: Ok I see.
But in that case do we need to run test coverage after integration tests?

Comment (jbern0rd) [reply build.gradle:160]: As it is implemented, there is no harm done.
When applying the **jacoco** plugin, all test tasks are instrumented by default to produce coverage data.
Here, we only configure the system to generate **coverage reports** after all test tasks have been executed.

Comment (zguesmi) [reply build.gradle:160]: Ok no problem. We can keep it that way.

**Review by zguesmi — APPROVED** [review]

**Pull request merged**